querier_test.go

```diff
@@ -1691,6 +1692,118 @@ func BenchmarkQuerySeek(b *testing.B) {
 	}
 }
 
+func BenchmarkSetMatcher(b *testing.B) {
+	cases := []struct {
+		numBlocks                   int
+		numSeries                   int
+		numSamplesPerSeriesPerBlock int
+		setPattern                  string
+		regexPattern                string
+	}{
+		{
+			numBlocks:                   1,
+			numSeries:                   15,
```

Comment:
As this is a benchmark, let's keep this number a little higher (a few thousands). Also a higher `numSamplesPerSeriesPerBlock`.

Reply:
This case I set 15 is because I want to check if the case "I'd be interested to know if there's a point where it's better to use the regex rather than the set." exists.

Reply:
Makes sense. You can also add benchmark case for higher number of series to know how it will impact real settings.